py/desispec/correct_cte.py

```diff
@@ -627,22 +697,20 @@ def get_cte_images(night, camera, expids=None):
         keep = exptable['OBSTYPE'] == 'flat'
         exptable = exptable[keep]
 
-        #- Find a 1sec and 120sec flat
-        selection = (np.abs(exptable['EXPTIME'] - 1) < 0.1)
-        if np.sum(selection)<1 :
-            mess = f"No good {camera} 1 sec flat for {night} (in {exptablefn}), needed for CTE correction model fit"
-            log.critical(mess)
-            raise RuntimeError(mess)
-        index1 = np.where(selection)[0][0]
-
-        selection = (np.abs(exptable['EXPTIME'] - 120) < 10)
-        if np.sum(selection)<1 :
-            mess = f"No good {camera} 120 sec flat for {night} (in {exptablefn}), needed for CTE correction model fit"
-            log.critical(mess)
-            raise RuntimeError(mess)
-        index2 = np.where(selection)[0][-1]
-
-        expids = list(exptable['EXPID'][ [index1, index2] ])
+        lengths = [1, 3, 10, 120]
+        indices = []
+        for length in lengths:
+            selection = (np.abs(exptable['EXPTIME'] - length) < 0.1) & (exptable['OBSTYPE'] == 'flat')
+            if np.sum(selection)<1:
+                if length not in [3, 10]:
+                    mess = f"No flat exposure of approx. {length} found for night {night} (in {exptablefn}). It's a requirement for the CTE correction model fit"
+                    log.error(mess)
+                    raise RuntimeError(mess)
+                else:
+                    continue
+            indices.append(np.where(selection)[0][0])
```

Comment:
Confirming logic here: 1s and 120s flats are still required for CTE modeling, and 3s and 10s are used if available but are optional?

Reply:
Correct.